From e79b0c9aa8a1027798d46c820f9079139ae46203 Mon Sep 17 00:00:00 2001
From: nibon7 <nibon7@gmail.com>
Date: Thu, 22 Aug 2024 18:18:04 +0800
Subject: [PATCH] shell-rs: Fix loading encrypted pdf

The new rust port of PpsWindow 3e2eeccdfa9c fails to load encrypted PDF
files, since the load job may be cleared before we unlock them.
---
 shell-rs/src/window.rs | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/shell-rs/src/window.rs b/shell-rs/src/window.rs
index e78429e62..b1356b786 100644
--- a/shell-rs/src/window.rs
+++ b/shell-rs/src/window.rs
@@ -917,6 +917,8 @@ mod imp {
                                     job.password_save(),
                                 );
                             }
+
+                            obj.clear_load_job();
                         }
                         Err(e) => {
                             if e.matches(papers_document::DocumentError::Encrypted)
@@ -949,12 +951,11 @@ mod imp {
                             } else {
                                 obj.show_error(Some(&e));
 
+                                obj.clear_load_job();
                                 obj.clear_local_uri();
                             }
                         }
                     }
-
-                    obj.clear_load_job();
                 }
             ));
 
-- 
2.44.2

